widget/src/providers/SocketProvider.tsx

```diff
@@ -14,41 +14,72 @@ import {
   useRef,
 } from "react";
 
+import { IOIncomingMessage } from "../types/io-message.types";
 import { getSocketIoClient, SocketIoClient } from "../utils/SocketIoClient";
 
 import { useConfig } from "./ConfigProvider";
+import { useCookie } from "./CookieProvider";
 
 interface socketContext {
   socket: SocketIoClient;
+  resetSocket: () => Promise<void>;
 }
 
 const socketContext = createContext<socketContext>({
   socket: {} as SocketIoClient,
+  resetSocket: async () => {},
 });
 
-export const SocketProvider = (props: PropsWithChildren) => {
+interface SocketProviderProps extends PropsWithChildren {
+  responseInterceptor?: <T>(
+    response: IOIncomingMessage<T>,
+  ) => Promise<IOIncomingMessage<T>>;
+}
+
+export const SocketProvider = (props: SocketProviderProps) => {
   const config = useConfig();
+  const { getCookie } = useCookie();
+  const defaultResponseInterceptor = async <T,>(
+    response: IOIncomingMessage<T>,
+  ) => {
+    if (response.statusCode === 401) {
+      await resetSocket();
+    }
+
+    return response;
+  };
   const socketRef = useRef(
-    getSocketIoClient(config, {
-      onConnect: () => {
-        // eslint-disable-next-line no-console
-        console.info(
-          "Hexabot Live Chat : Successfully established WS Connection!",
-        );
-      },
-      onConnectError: () => {
-        // eslint-disable-next-line no-console
-        console.error("Hexabot Live Chat : Failed to establish WS Connection!");
-      },
-      onDisconnect: () => {
-        // eslint-disable-next-line no-console
-        console.info("Hexabot Live Chat : Disconnected WS.");
+    getSocketIoClient(
+      config,
+      {
+        onConnect: () => {
+          // eslint-disable-next-line no-console
+          console.info(
+            "Hexabot Live Chat : Successfully established WS Connection!",
+          );
+        },
+        onConnectError: () => {
+          // eslint-disable-next-line no-console
+          console.error(
+            "Hexabot Live Chat : Failed to establish WS Connection!",
+          );
+        },
+        onDisconnect: () => {
+          // eslint-disable-next-line no-console
+          console.info("Hexabot Live Chat : Disconnected WS.");
+        },
       },
-    }),
+      props.responseInterceptor || defaultResponseInterceptor,
+    ),
   );
+  const resetSocket = async () => {
+    socketRef.current.disconnect();
+    await getCookie();
+    socketRef.current.connect();
+  };
```

Comment:
No need to have this part anymore, cookie will be returned on any socket connection, if the cookie/session expires, the API will set a new cookie.